guardian/amigo
# First draft of an AMI-level mitigation for CVE-2021-44228

## What does this change?

Add a role to set an environment variable that will (partially) disable the Log4J2 feature at the root of CVE-2021-44228.

We are adding this to a new role that we intend to add to all our base images. This ensures that this feature is globally enabled, to catch our Java AMIs as well as anything else that uses Java surprisingly.



## How to test

We'll have to bake an image and check that this environment variable is correctly configured on a box that is started up with the newly-built AMI. We'll then also check that an application started the normal way also sees the environment variable.



## Have we considered potential risks?

We'll need to make sure that this works before we can consider it a true mitigation but this is only a defence-in-depth approach to tackling the Log4J issue, so the risk here is small.
